keps/sig-apps/0003-kubernetes-application-api.md

```diff
@@ -0,0 +1,1108 @@
+
+#### Garbage Collection
+The Application controller may, in the future, implement OwnerReference propagation for Garbage Collection. However, 
+
+#### Installation Status Reporting
+The Application Controller will additionally update the `Status` of Application objects. When the Application Controller 
+detects newly created objects, for all objects other than Applications, it shall do the following.
+
+1. [Retrieve the Application for the object](#retrieving-applications).
+1. If the Application's `Selector` matches the object, the Application Controller shall update the Applications status 
+to indicate that component is installed.
+
+For all Application objects, the Application Controller shall do the following upon creation.
+
+1. [Select the Application's components](#selecting-components).
+1. For each selected object, the Application Controller will update the Application object's `Status` to indicate that 
+the component is installed. If a component is not found, the Applications status will be updated to indicate that the 
+component is not installed.
+1. [Retrieve the Application's dependencies](#retrieving-dependencies).
+1. For each dependency, the Application controller will update the the Application object's `Status` to indicate the 
+installation status of the dependency.
+1. The Application Controller will then scan all installed Applications in the same namespace as the Application 
+under consideration. If any of those Applications list the Application under consideration in their `Spec.Dependencies`, 
+their `Status` will be updated to indicate that the dependency has been installed.
+
+For all objects other than Applications, upon object deletion, the Application Controller shall do the following. 
+
+1. [Retrieve the Application for the object](#retrieving-applications).
+1. Update the Application's `Status` to indicate that component is no longer installed.
+
+For all Application objects, upon deletion, the Application Controller will do the following.
+
+1. List all Applications in the same namespace as the application under consideration.
+1. For all Applications that contain a reference to the Application under consideration in their `Depepndecies`, the 
+Application Controller will update the Application's `Statuts` to indicate that the Application under consideration is 
+no longer installed.
+
+For all objects other than Applications, when an object is updated, the Application Controller will do the following.
+
+1. [Retrieve the Application for the object](#retrieving-applications).
+1. If the Application's `Selector` does not match the object's labels, the Application Controller will update the 
+Applications `Status` to indicate that component is no longer installed and remove any `OwnerReference` from the 
+Application to the object.
+
+#### Health Status Reporting.
+The Application Controller will do the following to update the `Stauts.Ready` ConditionStatus of an Application object.
```

Comment:
What does this mean?   The app is ready to take full load as a service?  The App is ready to use by things that depend on it?